### PR TITLE
VideoIn

### PR DESCRIPTION
**Pull request by gumilastik**
Sources of modified Aforge library can be taken from my repo:
https://github.com/gumilastik/AForge.NET
(i don't know how to make symlinks to dlls).

Cannot add this project to vvvv-dx11.sln, beacuse this causes merge conflict.

### CODE REVIEW

File: Nodes/VVVV.DX11.Nodes.DShow/VideoIn.cs [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+			{
+				this.FTextureOutput[0].Dispose(context);
+			}
+		}
+		
+		public void Dispose()
+		{
+			StopCapture();
+			
+			this.FTextureOutput[0].Dispose();
```
Comment (mrvux) [inline Nodes/VVVV.DX11.Nodes.DShow/VideoIn.cs]: Needs a null check here

File: Nodes/VVVV.DX11.Nodes.DShow/VideoIn.cs [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+						this.FTextureOutput[0][context].WriteData(this.videoin.frontBuffer, this.videoin.GetSize());
+					}
+					else
+					{
+						this.FTextureOutput[0][context].WriteDataPitch(this.videoin.frontBuffer, this.videoin.GetSize());
+					}
+				}
+			}
+			
+			FOutIsNewFrame[0] = invalidateTexture;
```
Comment (mrvux) [inline Nodes/VVVV.DX11.Nodes.DShow/VideoIn.cs]: IsNewFrame should be in evaluate, as here it appears with one frame of delay

**Comment by mrvux** [comment]
Ok thanks, did not tried it yet, but just having a quick glance at the code could see following small issues.

**Comment by gumilastik** [comment]
Ok, i fixed it. Thanks for the tips! 
If you have time to do it properly threaded i will be very happy!

**Comment by mrvux** [comment]
Ok seems to work rather good here, I'll get that PR and add to solution (also I'll mod the output path to go in a split folder, which is my preferred way now).

**Comment by mrvux** [comment]
Ok all in there, and this commit https://github.com/mrvux/dx11-vvvv/commit/b7a94313cfd0f6da5f866d7c649fc72f32c3bc07 adds it to solution file

**Pull request merged**